apps/studio/src/lib/db/clients/bigquery.js

```diff
@@ -0,0 +1,276 @@
+  })
+
+  return {
+    async execute() {
+      // Get a query job first
+      [job] = await client.createQueryJob({ query: queryText })
+      logger().debug("job: ", job)
+
+      try {
+        logger().debug("wait for executeQuery job.id: ", job.id)
+        const data = await Promise.race([
+          cancelable.wait(),
+          executeQuery(client, queryText, job),
+        ])
+        return data
+      } catch (err) {
+        log.error('executeQuery error: ', err)
+        throw err
+      }
+    },
+    async cancel() {
+      if (!job) {
+        throw new Error('Query not ready to be canceled')
+      }
+      canceling = true
+      try {
+        const [jobCancelResponse] = await job.cancel()
+        logger().debug("query jobCancelResponse: ", jobCancelResponse)
+        cancelable.cancel()
+      } catch (err) {
+        canceling = false
+        throw err
+      } finally {
+        canceling = false
+        cancelable.discard()
+      }
+    },
+  }
+}
+
+
+function getVersionString() {
+  return null
+}
+
+
+function parseRowData(data) {
+  // BigQuery can return nested objects with custom types in the results
+  // look for the value string property.
+  // https://github.com/googleapis/nodejs-bigquery/blob/71dbed2140893677f7af254f5a7713a7f50bae92/src/bigquery.ts#L2191
+  return data.map((row) => {
+    const parsedRow = {}
+    Object.keys(row).forEach((key) => {
+      let strValue = row[key]
+      if (strValue != null && (Object.prototype.hasOwnProperty.call(strValue, 'value'))) {
+        strValue = row[key].value
+      }
+      parsedRow[key] = strValue
+    })
+    return parsedRow
+  })
+}
+
+
+function parseRowQueryResult(data) {
+  // Fallback in case the identifier could not reconize the command
+  const isSelect = Array.isArray(data)
+  const rows = parseRowData(data) || []
+  const fields = Object.keys(rows[0] || {}).map((name) => ({ name, id: name }))
+
+  logger().debug("parseRowQueryResult data: ", data)
+  return {
+    command: isSelect ? 'SELECT' : 'UNKNOWN',
+    rows,
+    fields: fields,
+    rowCount: data && data.length,
+    affectedRows: 0,
+  }
+}
+
+
+async function executeQuery(client, queries, job) {
+  // Support passing a single query string and an object with params
+  if (queries instanceof String) {
+    queries = { query: queries }
+  }
+  if (!job) {
+    [job] = await client.createQueryJob(queries)
+  }
+  // For Debugging increase execution time
+  // if (process.env.NODE_ENV === 'development') {
+  //   await new Promise(resolve => setTimeout(resolve, 5000))
+  // }
+
+  // Wait for the query to finish
+  const results = await job.getQueryResults()
+  return results.map(parseRowQueryResult)
+}
+
+
+export async function disconnect(client) {
+  return Promise.resolve()
+}
+
+
+function prettyJSON(obj) {
+  logger().debug(JSON.stringify(obj, null, 2))
+}
+
+
+async function listTablesOrViews(client, db, type) {
+  // Lists all tables or views in the dataset
```

Comment:
This throws an error if there is no dataset currently selected (connecting without a default dataset for instance.

Reply:
I have found the bug for this; this line is not needed in the implementation. 
```server.config.client = client // save the client for later use```